sql/sql_table.cc

```diff
@@ -9139,17 +9138,27 @@ fk_check_column_changes(THD *thd, Alter_info *alter_info,
         return FK_COLUMN_RENAMED;
       }
 
+      /*
+        Field_{num|decimal}::is_equal evaluates to IS_EQUAL_NO where
+        the new_field adds an AUTO_INCREMENT flag on a column due to a
+	limitation in MyISAM/ARIA. For the purposes of FK determination
+        it doesn't matter if AUTO_INCREMENT is there or not.
+      */
+      const auto flags= new_field->flags;
+      new_field->flags&= ~AUTO_INCREMENT_FLAG;
```

Comment:
Can you please reorganize the code this way:
```
      const uint flags= new_field->flags;
      new_field->flags&= ~AUTO_INCREMENT_FLAG;
      const bool equal_result= old_field->is_equal(*new_field);
      new_field->flags= flags;

      if ((equal_result == IS_EQUAL_NO) || ...
```

Then you won't need to reset flags twice.